bciers/libs/components/src/form/MultiStepBase.tsx

```diff
@@ -54,29 +55,54 @@ const MultiStepBase = ({
   uiSchema,
   submitButtonDisabled,
   customValidate,
+  successComponent,
 }: MultiStepBaseProps) => {
   const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isSuccessfullySubmitted, setIsSuccessfullySubmitted] = useState(false);
   const [isEditMode, setIsEditMode] = useState(false);
-
+  const [error, setError] = useState(parentError);
   const router = useRouter();
   const isNotFinalStep = step !== steps?.length;
 
   const stepIndex = step - 1;
 
-  // Set isSubmitting to true to disable submit buttons and prevent multiple form submissions
   const submitHandler = async (data: any) => {
+    // Set isSubmitting to true to disable submit buttons and prevent multiple form submissions
     setIsSubmitting(true);
-    const response = await onSubmit(data);
-
-    // If there is an error, set isSubmitting to false to re-enable submit buttons
-    // and allow user to attempt to re-submit the form
-    if (response?.error) {
+    // Clear any old errors
+    setError(undefined);
+    try {
+      const response = await onSubmit(data);
+      if (response?.error) {
+        setError(response?.error);
+        return;
+      }
+      // First step
+      if (step === 1) {
+        // The URL below is customized for the registration workflow. It can be generalized later if needed.
+        const nextStepUrl = `/register-an-operation/${response.id}/${step + 1}${
+          baseUrlParams ? `?${baseUrlParams}` : ""
+        }`;
+        router.push(nextStepUrl);
+        return;
+      }
+      // Middle steps
+      if (isNotFinalStep && baseUrl) {
+        const nextStepUrl = `${baseUrl}/${step + 1}${
+          baseUrlParams ? `?${baseUrlParams}` : ""
+        }`;
+        router.push(nextStepUrl);
+        return;
+      }
+      // Last step
+      if (!isNotFinalStep) {
+        setIsSuccessfullySubmitted(true);
+      }
```

Comment:
This is supposed to be a general use form base component, we shouldn't have routes such as `register-an-operation` hardcoded in it. I get that this is trying to reduce duplication but I find this kind of confusing and it makes this component less reusable. I'm definitely pro automating a lot of this though some things make sense to handle in each form component.

Suggest keeping `baseUrl` prop required and remove most of this refactor except perhaps the last step block.

Reply:
As discussed, the first step does need to be handled differently, I went with a prop.